linksight/api/models.py

```diff
@@ -105,12 +137,15 @@ def generate_match_items(self, **kwargs):
         ]
 
         matched_raw = pd.DataFrame()
+        matches_json = []
         for index, row in dataset_df.iterrows():
```

Comment:
Instead of matching per row, is it possible to insert the source fields to be matched to a new model (say Source) and run a single query?